scripts/tests/test_generate_bad_phase37_artifacts.py

```
@@ -0,0 +1,158 @@
+            )
+
+    def test_generated_artifacts_match_expected_reference_verifier_outcomes(self) -> None:
+        with tempfile.TemporaryDirectory() as tempdir:
+            output_dir = pathlib.Path(tempdir) / "mutations"
+            manifest = load_json(run_generator(output_dir))
+            by_name = {item["name"]: item for item in manifest["mutations"]}
+
+            for item in manifest["mutations"]:
+                artifact = load_json(output_dir / item["file_name"])
+                if item["expected_reference_verifier"] == "fail":
+                    with self.assertRaisesRegex(REF.ReferenceVerifierError, re.escape(item["actual_error"][:40])):
```

Ruff (0.15.10)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)